llvm/lib/Transforms/Scalar/GVN.cpp

```diff
@@ -887,7 +892,8 @@ PreservedAnalyses GVNPass::run(Function &F, FunctionAnalysisManager &AM) {
   auto *MemDep =
       isMemDepEnabled() ? &AM.getResult<MemoryDependenceAnalysis>(F) : nullptr;
   auto &LI = AM.getResult<LoopAnalysis>(F);
-  auto *MSSA = AM.getCachedResult<MemorySSAAnalysis>(F);
+  auto *MSSA =
+      isMemorySSAEnabled() ? AM.getCachedResult<MemorySSAAnalysis>(F) : nullptr;
```

Comment:
Should avoid retrieving cached results (and thus computing MemorySSAUpdater) if MemorySSA is not enabled (missed from the initial patch, realized while running tests, may happen that both MD and MSSAU exist, as of now).